lib/Mojolicious/Guides/Growing.pod

```diff
@@ -0,0 +1,776 @@
+
+=encoding utf8
+
+=head1 NAME
+
+Mojolicious::Guides::Growing - Growing Mojolicious applications
+
+=head1 OVERVIEW
+
+This document explains the process of starting a L<Mojolicious::Lite> prototype from scratch and growing it into a
+well-structured L<Mojolicious> application. The final result of this guide is also available as an
+L<example application|https://github.com/mojolicious/mojo/tree/main/examples/login>.
+
+=head1 CONCEPTS
+
+Essentials every L<Mojolicious> developer should know.
+
+=head2 Model View Controller
+
+MVC is a software architectural pattern for graphical user interface programming originating in Smalltalk-80, that
+separates application logic, presentation and input.
+
+           +------------+    +-------+    +------+
+  Input -> | Controller | -> | Model | -> | View | -> Output
+           +------------+    +-------+    +------+
+
+A slightly modified version of the pattern moving some application logic into the I<controller> is the foundation of
+pretty much every web framework these days, including L<Mojolicious>.
+
+              +----------------+     +-------+
+  Request  -> |                | <-> | Model |
+              |                |     +-------+
+              |   Controller   |
+              |                |     +-------+
+  Response <- |                | <-> | View  |
+              +----------------+     +-------+
+
+The I<controller> receives a request from a user, passes incoming data to the I<model> and retrieves data from it,
+which then gets turned into an actual response by the I<view>. But note that this pattern is just a guideline that most
+of the time results in cleaner more maintainable code, not a rule that should be followed at all costs.
+
+=head2 REpresentational State Transfer
+
+REST is a software architectural style for distributed hypermedia systems such as the web. While it can be applied to
+many protocols it is most commonly used with HTTP these days. In REST terms, when you are opening a URL like
+C<http://mojolicious.org/foo> with your browser, you are basically asking the web server for the HTML I<representation>
+of the C<http://mojolicious.org/foo> I<resource>.
+
+  +--------+                                  +--------+
+  |        | -> http://mojolicious.org/foo -> |        |
+  | Client |                                  | Server |
+  |        | <-  <html>Mojo rocks!</html>  <- |        |
+  +--------+                                  +--------+
+
+The fundamental idea here is that all resources are uniquely addressable with URLs and every resource can have
+different representations such as HTML, RSS or JSON. User interface concerns are separated from data storage concerns
+and all session state is kept client-side.
```

Comment:
All?  Necessarily?  I think always at minimum one piece of data is kept client side in the form of a cookie, but _all_ of the session state might still be kept server side.  Unless this technical possibility is counter to REST?